src/trait_/common_ord.rs

```diff
@@ -49,114 +57,153 @@ pub fn build_ord_signature(item: &Item, trait_: &DeriveTrait, body: &TokenStream
 				}
 			};
 
-			// Nightly or unsafe (default) implementation.
-			#[cfg(any(feature = "nightly", not(feature = "safe")))]
-			{
-				let path = trait_.path();
-				let method = match trait_ {
-					PartialOrd => quote! { partial_cmp },
-					Ord => quote! { cmp },
-					_ => unreachable!("unsupported trait in `prepare_ord`"),
-				};
+			let incomparable = build_incomparable_pattern(variants);
 
-				// Nightly implementation.
-				#[cfg(feature = "nightly")]
+			// If there is only one comparable variant, it has to be it when it is non
+			// incomparable.
+			let mut comparable = variants.iter().filter(|v| !v.is_incomparable());
+			if let (Some(comparable), None) = (comparable.next(), comparable.next()) {
+				let incomparable = incomparable.expect("there are > 1 variants");
```

Comment:
```suggestion
				let incomparable = incomparable.expect("there should be > 1 variants");
```
The error message is otherwise quite confusing.
https://doc.rust-lang.org/nightly/std/error/index.html#common-message-styles